solidity/random-beacon/tasks/initialize.ts

```diff
@@ -0,0 +1,285 @@
+  .addParam("owner", "Stake Owner address", undefined, types.string)
+  .addParam("provider", "Staking Provider", undefined, types.string)
+  .addOptionalParam("beneficiary", "Stake Beneficiary", undefined, types.string)
+  .addOptionalParam("authorizer", "Stake Authorizer", undefined, types.string)
+  .addOptionalParam("amount", "Stake amount", 1_000_000, types.int)
+  .setAction(async (args, hre) => {
+    await stake(hre, args)
+  })
+
+async function stake(
+  hre: HardhatRuntimeEnvironment,
+  args: {
+    owner: string
+    provider: string
+    beneficiary: string
+    authorizer: string
+    amount: BigNumber
+  }
+) {
+  const { ethers, helpers } = hre
+  const { owner, provider } = args
+
+  // Beneficiary can equal to the owner if not set otherwise. This simplification
+  // is used for development purposes.
+  const beneficiary = args.beneficiary ?? owner
+
+  // Authorizer can equal to the owner if not set otherwise. This simplification
+  // is used for development purposes.
+  const authorizer = args.authorizer ?? owner
+
+  const { to1e18, from1e18 } = helpers.number
+  const t = await helpers.contracts.getContract("T")
+  const staking = await helpers.contracts.getContract("TokenStaking")
+
+  const stakedAmount = to1e18(args.amount)
+
+  console.log(
+    `Staking ${from1e18(stakedAmount)} T to the staking provider ${provider}...`
+  )
+
+  await (
+    await staking
+      .connect(await ethers.getSigner(owner))
+      .stake(provider, beneficiary, authorizer, stakedAmount)
+  ).wait()
+}
+
+task(TASK_AUTHORIZE, "Sets authorization")
+  .addParam("owner", "Stake Owner address", undefined, types.string)
+  .addParam("provider", "Staking Provider", undefined, types.string)
+  .addOptionalParam("beneficiary", "Stake Beneficiary", undefined, types.string)
+  .addOptionalParam(
+    "authorization",
+    "Authorization amount (default: minimumAuthorization)",
+    undefined,
+    types.int
+  )
+  .setAction(async (args, hre) => {
+    await authorize(hre, args)
+  })
+
+async function authorize(
+  hre: HardhatRuntimeEnvironment,
+  args: {
+    owner: string
+    provider: string
+    authorizer: string
+    authorization: BigNumber
+  }
+) {
+  const { ethers, helpers } = hre
+  const { owner, provider } = args
+
+  const randomBeacon = await helpers.contracts.getContract("RandomBeacon")
+
+  // Authorizer can equal to the owner if not set otherwise. This simplification
+  // is used for development purposes.
+  const authorizer = args.authorizer ?? owner
+
+  const { to1e18, from1e18 } = helpers.number
+  const t = await helpers.contracts.getContract("T")
+  const staking = await helpers.contracts.getContract("TokenStaking")
+
+  const authorization = args.authorization
+    ? to1e18(args.authorization)
+    : await randomBeacon.minimumAuthorization()
+
+  const currentAuthorization = await staking.authorizedStake(
+    provider,
+    randomBeacon.address
+  )
+
+  if (currentAuthorization.gte(authorization)) {
```

Comment:
Also, I don't think it makes sense to check `if (currentAuthorization.gte(authorization))` because on line 43 we `return` if the `currentAuthorization` is anything other than 0. If we treat this as a separate task that can be invoked independently, then we should probably use the same logic and return if the `currentAuthorization` is other than 0. Or we need to change `isAlreadyStaked()` to align to the logic in `TASK_STAKE`.

Reply:
> Just double checking.. I thought that `authorization` is an increased amount no matter what the current value is and we add this number to the current authorization amount. You sure it is supposed to work like you suggested? https://github.com/threshold-network/solidity-contracts/blob/main/contracts/staking/TokenStaking.sol#L508

The `TokenStaking.increaseAuthorization` function we use increments the current authorization. But from the initialization scripts perspective, we want to set up authorization to a certain level. Imagine that we run initialization in testnet node provisioning setup. We don't want to bump the authorization each time the node starts, instead, we want to set it to a specific minimum value.

Reply:
ok, but don't we want to have the same logic in `isAlreadyStake` and `TASK_AUTHORIZE`? Say `currentAuthorization = 1

in `isAlreadyStake`:

```
  (...)
  if (currentAuthorization === 0) {
    return false
  }

  return true // if currentAuthorization=1 we return true
```

in `TASK_AUTHORIZE`:
```
  (...)
  if (currentAuthorization.gte(authorization)) {
    console.log(
      `Authorized stake for the Random Beacon is ${from1e18(
        currentAuthorization
      )} T`
    )
    return
  }

  const increaseAmount = authorization.sub(currentAuthorization)
```
we do not return but we increase authorization. 

Two places where we get `const currentAuthorization = await staking.authorizedStake()` treat `currentAuthorization` differently.

Reply:
I improved the whole solution a219a066115116aa92bb2dd1d73c5aa7cefc815d